src/routes/user.ts

```
@@ -0,0 +1,79 @@
+import express from 'express'
+import { User, UserDocument } from './../db/schema/user'
+import bcrypt from 'bcrypt'
+import jwt from 'jsonwebtoken'
+
+const router = express.Router()
+
+router.post('/signup', async (req, res) => {
+
+    const userExists = await User.find({
+        email: req.body.email
+    }).exec()
+
+    if (userExists.length != 0) {
+        return res.status(400).json("User already exists")
+    }
+
+    const salt = await bcrypt.genSalt(10)
+    const password = await bcrypt.hash(req.body.password, salt)
```

Bcrypt will only use the first 72 bytes of input, silently ignoring excess bytes. Consider limiting password length to 72 bytes, because longer passwords provide no additional security.